From eb7e16f329adb3142a30a4a23ece36cf13be8caa Mon Sep 17 00:00:00 2001
From: Denis Barbier <bouzim@gmail.com>
Date: Sun, 12 Aug 2012 13:48:09 +0200
Subject: [PATCH] Drop getGL2PSFontName method from OpenGl_Display.hxx

Having member functions depending upon cpp macros is a bad idea,
it may cause crash if OCE and 3rd party code do not define
HAVE_GL2PS in a consistent manner.

Bump OCE_ABI_SOVERSION.
---
 CMakeLists.txt                  |    2 +-
 inc/OpenGl_Display.hxx          |    4 ----
 src/OpenGl/OpenGl_Display_1.cxx |    2 +-
 3 files changed, 2 insertions(+), 6 deletions(-)

Index: oce/inc/OpenGl_Display.hxx
===================================================================
--- oce.orig/inc/OpenGl_Display.hxx
+++ oce/inc/OpenGl_Display.hxx
@@ -133,10 +133,6 @@ class OpenGl_Display : public MMgt_TShar
 
   void ExportText (const wchar_t *text, const int is2d, const float x, const float y, const float z, const OpenGl_AspectText *aspect, const OpenGl_TextParam *param, const short height);
 
-#ifdef HAVE_GL2PS
-  static void getGL2PSFontName(const char *src_font, char *ps_font);
-#endif
-
   Aspect_Display   myDisplay;
   OpenGl_Facilities myFacilities;
 
Index: oce/src/OpenGl/OpenGl_Display_1.cxx
===================================================================
--- oce.orig/src/OpenGl/OpenGl_Display_1.cxx
+++ oce/src/OpenGl/OpenGl_Display_1.cxx
@@ -90,7 +90,7 @@ static const FontMapNode myFontMap[] =
 */
 
 #ifdef HAVE_GL2PS
-void OpenGl_Display::getGL2PSFontName (const char *src_font, char *ps_font)
+static void getGL2PSFontName (const char *src_font, char *ps_font)
 {
   /* 
   Convert font name used for rendering to some "good" font names
